### PR TITLE
*: skip creating Domain for backup without stats (#876)

### LINKED ISSUE #818: `BuildBackupRangeAndSchema` leaks memory
1. What did you do?
If possible, provide a recipe for reproducing the error.

BR backup holds too much memory if a cluster has many tables.

We take heap profile during backup, they show `BuildBackupRangeAndSchema` always hold 570MB memory even if the function returned already.

https://gist.github.com/overvenus/4491e987dca672fd62705d29bb254d03

2. What did you expect to see?

`BuildBackupRangeAndSchema` should not leak, and it's memory consumption should be O(1) not O(n) (n is a number of tables). 

3. What did you see instead?

BR consumes too much memory.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #876 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/886
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/886:release-4.0-de9666974ff3
```

---

### What problem does this PR solve? 

This PR reduces memory usage during backup by skipping initializing Domain which loads all table info into memory.

Close #818 

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

<details> <summary> Manual tests </summary>

1. Load 40k 40 fields YCSB tables

```
for i in $(seq 1 40000); do { go-ycsb load mysql -p workload=core -p mysql.host=172.16.5.33 -p mysql.port=47904 -p mysql.user=root -p recordcount=20 -p fieldcount=40 -p fieldlength=16 -p threadcount=4 -p table=usertable$i; } done;
```

2. Backup 40k tables

```
br --pd http://PD_ADDR backup full -s "local:///path/to/backupdir"
```

</details>

The master branch BR takes about 3.09GB of memory.

This PR BR takes about 2.21G of memory.

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - Reduce memory usage during backup.

### CODE REVIEW

File: pkg/task/backup.go [hunk 1]
```diff
@@ -266,8 +275,12 @@ func RunBackup(c context.Context, g glue.Glue, cmdName string, cfg *BackupConfig
 		return errors.Trace(err)
 	}
 
+<<<<<<< HEAD
 	ranges, backupSchemas, err := backup.BuildBackupRangeAndSchema(
 		mgr.GetDomain(), mgr.GetTiKV(), cfg.TableFilter, backupTS, cfg.IgnoreStats)
+=======
+	ranges, schemas, err := backup.BuildBackupRangeAndSchema(mgr.GetStorage(), cfg.TableFilter, backupTS)
+>>>>>>> de96669... *: skip creating Domain for backup without stats (#876)
```
Comment (kennytm) [inline pkg/task/backup.go:283]: merge conflict

**Comment by kennytm** [comment]
/run-integration-test

stats stuff

**Review by overvenus — APPROVED** [review]

**Pull request merged**